lambda_uploader/package.py

```diff
@@ -24,83 +24,119 @@
 LOG = logging.getLogger(__name__)
 TEMP_WORKSPACE_NAME = ".lambda_uploader_temp"
 ZIPFILE_NAME = 'lambda_function.zip'
-MAX_PACKAGE_SIZE = 50000000
-
-
-def build_package(
-    path,
-    requirements,
-    virtualenv=None,
-    ignore=[],
-    zipfile_name=ZIPFILE_NAME
-):
-    pkg = Package(path, virtualenv, requirements, zipfile_name)
-
-    pkg.clean_workspace()
-    pkg.clean_zipfile()
-    pkg.prepare_workspace()
-    pkg.prepare_virtualenv()
-    pkg.package(ignore)
-    pkg.check_current_package_size()
+
+
+def build_package(path, requirements, virtualenv=None, ignore=[],
+                  zipfile_name=ZIPFILE_NAME):
+    pkg = Package(path, zipfile_name)
+
+    if virtualenv is not None:
+        pkg.virtualenv(virtualenv)
+    pkg.requirements(requirements)
+    pkg.build(ignore)
+
     return pkg
 
 
 class Package(object):
-    def __init__(
-        self,
-        path,
-        virtualenv=None,
-        requirements=[],
-        zipfile_name=ZIPFILE_NAME
-    ):
+    def __init__(self, path, zipfile_name=ZIPFILE_NAME):
         self._path = path
         self._temp_workspace = os.path.join(path,
                                             TEMP_WORKSPACE_NAME)
+
         self.zip_file = os.path.join(path, zipfile_name)
-        self._virtualenv = virtualenv
-        self._requirements = requirements
+        self._virtualenv = None
+        self._skip_virtualenv = False
```

Comment:
Before this PR, I think `_virtualenv = False` meant 'skip this step,' but now we have a new variable that means the same thing. Not sure we need it, but it isn't hurting anything.